### ISSUE TITLE
feature: Show available trailmark stacks

### ISSUE THREAD

**Issue by FelixKratz** [LeonHeidelbach/trailblazer.nvim#26]
### Did you check the docs?

- [X] I have read all the trailblazer.nvim docs

### Is your feature request related to a problem? Please describe.

I would like to have a visual indication which trail mark stacks are available and which one is selected.

### Describe the solution you'd like

Something like this in the first row of the qf list perhaps:
![Screenshot 2023-02-27 at 11 48 56](https://user-images.githubusercontent.com/22680421/221544538-253be8ac-a6af-4a8f-a401-ef8f04b0558b.png)

### Describe alternatives you've considered

--

Edit: I know that there is this quickfix title:
![Screenshot 2023-02-27 at 12 01 41](https://user-images.githubusercontent.com/22680421/221547009-31ab5a11-3750-4c03-a530-8c55f692032a.png)
but it also does not show the available stacks.

**Comment by LeonHeidelbach** [comment]
@FelixKratz You can already see all available trail mark stacks in the tab completion menu when using the corresponding user command. This is unfortunately something that cannot be implemented in terms of a quickfix list but would require a custom implementation. However, I think the tab completion menu should be good enough for this use case.

![Screenshot_20230227_143135](https://user-images.githubusercontent.com/55194243/221576771-07daef19-a8e7-4eab-b1e2-305792120c08.png)

**Comment by FelixKratz** [comment]
I dont use the tab completion menu since I dont use the user command, rather I have a mapping like this:
```lua
vim.api.nvim_set_keymap("n", "<Esc>as", ':lua local name = "" vim.ui.input({ prompt = "Stack Name: " }, function(input) name = input end) require("trailblazer").switch_trail_mark_stack(name)<CR>', opts)
```
which I use for switching and creating stacks.

A way to see all available stacks in the quickfix title or the first row would be really helpful for navigating with
previous and next stack commands as well:
```lua
vim.api.nvim_set_keymap("n", "<Esc>p", ':lua require("trailblazer").switch_to_previous_trail_mark_stack()<CR>', opts)
vim.api.nvim_set_keymap("n", "<Esc>n", ':lua require("trailblazer").switch_to_next_trail_mark_stack()<CR>', opts)
```

**Comment by LeonHeidelbach** [comment]
> I dont use the tab completion menu since I dont use the user command, rather I have a mapping like this:
> 
> ```lua
> vim.api.nvim_set_keymap("n", "<Esc>as", ':lua local name = "" vim.ui.input({ prompt = "Stack Name: " }, function(input) name = input end) require("trailblazer").switch_trail_mark_stack(name)<CR>', opts)
> ```
> 
> which I use for switching and creating stacks.

Why don't you just create your keymap like this? Your map seems a bit overcomplicated since the functionality you implemented basically already exists with the user command itself and it also has tab completion.

```lua
vim.api.nvim_set_keymap("n", "<Esc>as", ':TrailBlazerSwitchTrailMarkStack ', opts)
```

Notice the missing `<CR>` at the end. This requires you to manually press <kbd>Enter</kbd> just like with your implementation.

> A way to see all available stacks in the quickfix title or the first row would be really helpful for navigating with previous and next stack commands as well:
> 
> ```lua
> vim.api.nvim_set_keymap("n", "<Esc>p", ':lua require("trailblazer").switch_to_previous_trail_mark_stack()<CR>', opts)
> vim.api.nvim_set_keymap("n", "<Esc>n", ':lua require("trailblazer").switch_to_next_trail_mark_stack()<CR>', opts)
> ```

An idea would be to maybe adjust the switch prompt like this, what do you think?

```console
TrailBlazer(Info): Trail mark stack switched: default << [Notes] >> Todo
```
`Notes` would then be the trail mark stack you switched to, left would be the previous and right would be the next.

**Comment by FelixKratz** [comment]
> An idea would be to maybe adjust the switch prompt like this, what do you think?
> 
> ```
> TrailBlazer(Info): Trail mark stack switched: default << [Notes] >> Todo
> ```
> 
> `Notes` would then be the trail mark stack you switched to, left would be the previous and right would be the next.

The problem with this is that I only see previous and next *after* I already did a switch. Thus I only gain info about the stack where I came from (which I know) and the next stack. It is still possible that the stack I want to navigate to was prior to the first stack and I went in the wrong direction and have to cycle through all stacks.

Apart from that when I have multiple stacks I want to know how they are ordered such that I can decide wheter I have to press next or prev (and how often) before starting navigation

**Comment by LeonHeidelbach** [comment]
> The problem with this is that I only see previous and next _after_ I already did a switch. Thus I only gain info about the stack where I came from (which I know) and the next stack. It is still possible that the stack I want to navigate to was prior to the first stack and I went in the wrong direction and have to cycle through all stacks.
> 
> Apart from that when I have multiple stacks I want to know how they are ordered such that I can decide wheter I have to press next or prev (and how often) before starting navigation

This is a very specific case in which I would actually recommend you not to use the switch commands but the key map I just proposed and use the tab completion list to directly find the stack you want to navigate to.

The only way I otherwise see this working is with an implementation like Harpoon's floating buffer list, which would somewhat look like this:

![Screenshot_20230227_163347](https://user-images.githubusercontent.com/55194243/221607640-0e0f84ba-d0bc-4213-a301-28d42b5d985d.png)

**Comment by FelixKratz** [comment]
I dont think that this is very specific, it is the same workflow most people use for their buffers with a bufferline rather than switching windows via some tab completion

**Comment by FelixKratz** [comment]
#31 Would solve this

**Comment by LeonHeidelbach** [comment]
> #31 Would solve this when the callback function is called with current stack and all available stacks. Then I would set it to show in my status bar

This has been implemented 3f5a83f28f5f4b02bb31922705e527d1661a8cf2. Closing this for now but different ways of showing the list of available trail mark stacks as detailed above might be implemented in the future.

**Comment by FelixKratz** [comment]
Works perfectly and is a must have for me:

https://user-images.githubusercontent.com/22680421/221832535-d7809696-2d5d-40e2-b361-f0d45f80ffa7.mp4
